### DISCUSSION TITLE
Vulnerability in PostreSQL - CVE-2024-10979

### DISCUSSION THREAD

**Discussion by JeremyQtweb** [supabase/supabase#30872] [Questions] [answered]
Hey Supabase team,

Wondering if there's a specific place we can find information about things like PostgreSQL vulnerabilities and what the Supabase team are currently doing regarding them.

We have been alerted to [CVE-2024-10979](https://nvd.nist.gov/vuln/detail/CVE-2024-10979) for example, and people higher than me are looking for some information about Supabase's position on it. Clearly to identify the level of risk such a vulnerability poses.

**Comment by GaryAustin1** [comment, marked as answer]
I'm not an employee but a quick glance shows that is about a Python language Postgres function extension...
https://github.com/orgs/supabase/discussions/2239 specifically mentions that extension and why Supabase won't install it on their systems.

Also the links refer to Postgres 16 and higher if I read them correctly and at the moment Supabase is still on 15.

But if you want an official response you should contact support directly.

**Reply by JeremyQtweb** [reply]
@GaryAustin1 - Thanks for this info, I did not make that connection. Nice work.

The vulnerability I linked describes the version affected as thus:

_Versions **before** PostgreSQL 17.1, 16.5, 15.9, 14.14, 13.17, and 12.21 are affected._

I read that as: If Supabase is running on any of these minor versions (or higher) for it's respective major version, then a fix has been applied. The current - maximum - version of PostgreSQL available in Supabase is 15.6.1.143. Hmm.

However, I see where you got you're versioning logic. It's a tad ambiguous to be sure but I suppose the crux of it is, if you cannot/don't install the extension, you're not exposed to this vulnerability.
